### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by hpesonen**
#### Summary:
Some files related to automatic RTD-documentation generation were outdated and now updated.

#### Please make sure

- [ ] You have updated the CHANGELOG.rst
- [ ] You have provided a short summary of your changes (see previous section)
- [ ] You have listed the copyright holder for the work you are submitting (see next section)

If your contribution adds, removes or somehow changes the functional behavior of the package, please check that

- [ ] You have included or updated all the relevant documentation 
- [ ] You have added appropriate unit tests to ensure the new features behave as expected

and the proposed changes pass all unit tests (check step 6 of CONTRIBUTING.rst for details)

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD3 (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -135,9 +135,9 @@ Before you submit a pull request, check that it meets these guidelines:
    Github Actions.
 2. If the pull request adds functionality, the docs should be updated. Put
    your new functionality into a function with a docstring, and add the
-   feature to the list in README.md.
-3. The pull request should work for Python 3.7, or later. Check
-   https://github.com/elfi-dev/elfi/actions/workflows/pytest.yml
+   feature to the list in README.rst.
```
Comment (uremes) [inline CONTRIBUTING.rst:138]: here `README.md` has been updated to `README.rst` but i think `README.md` was correct unless the intention was that the contributor is asked to update `index.rst` that has the same list in docs?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -135,9 +135,9 @@ Before you submit a pull request, check that it meets these guidelines:
...
+3. The pull request should work for Python 3.5 and later. Check
```
Comment (uremes) [inline CONTRIBUTING.rst:139]: i think this should be Python 3.7

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -135,9 +135,9 @@ Before you submit a pull request, check that it meets these guidelines:
...
+   https://travis-ci.org/elfi-dev/elfi/pull_requests
```
Comment (uremes) [inline CONTRIBUTING.rst:140]: i think the link in our current version is correct: https://github.com/elfi-dev/elfi/actions/workflows/pytest.yml

File: CONTRIBUTING.rst (hunk 1 above)
Comment (hpesonen) [reply CONTRIBUTING.rst:138]: Hmmm, this and the other issues you caught seem to indicate that I have updated the old versions of the documentation. Will fix them now.

**Review by uremes — APPROVED** [review]

**Pull request merged**